cylc/flow/network/server.py

```diff
@@ -148,89 +149,122 @@ def __init__(self, schd, context=None, barrier=None,
+        )
 
-    def _bespoke_start(self):
-        """Setup start items, and run listener.
+        self.replier = WorkflowReplier(self, context=self.zmq_context)
+        self.publisher = WorkflowPublisher(self, context=self.zmq_context)
 
-        Overwrites Base method.
+    def start(self, barrier):
+        """Start the TCP servers."""
+        # set asyncio loop on thread
+        try:
+            self.loop = asyncio.get_running_loop()
+        except RuntimeError:
+            self.loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(self.loop)
+
+        min_, max_ = glbl_cfg().get(['scheduler', 'run hosts', 'ports'])
+        self.replier.start(min_, max_)
+        self.publisher.start(min_, max_)
+        self.port = self.replier.port
+        self.pub_port = self.publisher.port
+        self.schd.data_store_mgr.delta_workflow_ports()
+
+        # wait for threads to setup socket ports before continuing
+        barrier.wait()
+
+        self.stopped = False
+
+        self.operate()
+
+    async def stop(self, reason):
+        """Stop the TCP servers, and clean up authentication."""
+        self.queue.put('STOP')
+        if self.thread and self.thread.is_alive():
+            while not self.stopping:
+                sleep(self.STOP_SLEEP_INTERVAL)
+
+        if self.replier:
+            self.replier.stop(stop_loop=False)
+        if self.publisher:
+            await self.publisher.publish(
+                [(b'shutdown', str(reason).encode('utf-8'))]
+            )
+            self.publisher.stop(stop_loop=False)
+        if self.curve_auth:
+            self.curve_auth.stop()  # stop the authentication thread
+        if self.loop and self.loop.is_running():
+            self.loop.stop()
+        if self.thread and self.thread.is_alive():
+            self.thread.join()  # Wait for processes to return
+
+        self.stopped = True
+
+    def operate(self):
+        """Orchestrate the receive, send, publish of messages."""
+        while True:
+            # process messages from the scheduler.
+            if self.queue.qsize():
+                message = self.queue.get()
+                if message == 'STOP':
+                    self.stopping = True
+                    break
+                raise ValueError('Unknown message "%s"' % message)
 
-        """
-        # start accepting requests
-        self.queue = Queue()
-        self.register_endpoints()
-        self._listener()
+            # Gather and respond to any requests.
+            self.replier.listener()
 
-    def _bespoke_stop(self):
-        """Stop the listener and Authenticator.
+            # Publish all requested/queued.
+            while self.publish_queue.qsize():
+                articles = self.publish_queue.get()
+                self.loop.run_until_complete(self.publisher.publish(articles))
 
-        Overwrites Base method.
+            # Yield control to other threads
+            sleep(self.OPERATE_SLEEP_INTERVAL)
```

Comment:
As above
```suggestion
            await asyncio.sleep(self.OPERATE_SLEEP_INTERVAL)
```

Reply:
`time.sleep` is used to yield control to other threads (halting the current thread).. Whereas, I believe `asyncio.sleep` just allows other ayncio execution for the duration.